src/notifications/common.ts

```diff
@@ -481,18 +481,22 @@ export const generateUserNotificationUniqueKey = ({
   type,
   referenceId,
   referenceType,
+  dedupKey,
 }: {
   type: NotificationType;
   referenceId?: string;
   referenceType?: NotificationReferenceType;
+  dedupKey?: string;
 }): string | null => {
   const uniqueKey = notificationTypeToUniqueKey[type];
 
   if (!uniqueKey) {
     return null;
   }
 
-  return [uniqueKey, referenceId, referenceType].filter(Boolean).join(':');
+  return [uniqueKey, referenceId, referenceType, dedupKey]
+    .filter(Boolean)
+    .join(':');
```

Comment:
since you keep referenceId, that would still be different post ids no and then it would generate multiple notifications? 

Best to add notification specific tests and verify it works

Reply:
still missing post notification tests for dedup logic?

Reply:
I've added this case https://github.com/dailydotdev/daily-api/pull/3154/files#diff-6e46397f1e2cc9f81f6d251217c5f625accc5596fbd5e11dcdb644922166ddb9R3436 , do you think we need to add more?